python/lsst/decam/hits/selectImages.py

```diff
@@ -0,0 +1,101 @@
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import lsst.pex.config as pexConfig
+import lsst.pex.exceptions as pexExceptions
+import lsst.afw.geom as afwGeom
+import lsst.pipe.base as pipeBase
+from lsst.pipe.tasks.selectImages import WcsSelectImagesTask, BaseExposureInfo
+
+__all__ = ["MaxPsfWcsSelectImagesTask"]
+
+
+class MaxPsfWcsSelectImageConfig(WcsSelectImagesTask.ConfigClass):
+    maxPsfFwhm = pexConfig.Field(dtype=float, doc="Maximum PSF FWHM (in pixels) to warp", default=5.)
+    minPsfFwhm = pexConfig.Field(dtype=float, doc="Minimum PSF FWHM (in pixels) to warp", default=0.)
+
+
+class MaxPsfWcsSelectImagesTask(WcsSelectImagesTask):
+    """Select images using their Wcs and a maximum seeing"""
+    ConfigClass = MaxPsfWcsSelectImageConfig
+
+    def runDataRef(self, dataRef, coordList, makeDataRefList=True, selectDataList=[]):
+        """Select images in the selectDataList that overlap the patch
+
+        We use the "convexHull" function in the geom package to define
+        polygons on the celestial sphere, and test the polygon of the
+        patch for overlap with the polygon of the image.
+
+        We use "convexHull" instead of generating a SphericalConvexPolygon
+        directly because the standard for the inputs to SphericalConvexPolygon
+        are pretty high and we don't want to be responsible for reaching them.
+        If "convexHull" is found to be too slow, we can revise this.
+
+        @param dataRef: Data reference for coadd/tempExp (with tract, patch)
+        @param coordList: List of Coord specifying boundary of patch
+        @param makeDataRefList: Construct a list of data references?
+        @param selectDataList: List of SelectStruct, to consider for selection
+        """
+        from lsst.geom import convexHull
```

Comment:
I think local imports are discouraged by PEP8?

Reply:
This looks totally standard to me, and I don't find anything relevant in PEP8--can you point me to it?

Reply:
I think I was thinking of the bit in https://www.python.org/dev/peps/pep-0008/#imports that says "Imports are always put at the top of the file", and the analogous text in the DM style guide. It's not explicitly forbidding local imports, but implies they should be kept together.

Reply:
Ah, I understand now.  Fixed.